Auth0/SafariAuthenticationCallback.swift

```diff
@@ -22,22 +22,45 @@
 
 import UIKit
 import SafariServices
+#if canImport(AuthenticationServices)
+import AuthenticationServices
+#endif
 
 #if swift(>=3.2)
 @available(iOS 11.0, *)
 class SafariAuthenticationSessionCallback: AuthTransaction {
 
     var state: String?
-    var authSession: SFAuthenticationSession?
     var callback: (Bool) -> Void = { _ in }
 
+    private var authSession: NSObject?
+
     init(url: URL, schemeURL: String, callback: @escaping (Bool) -> Void) {
         self.callback = callback
-        self.authSession = SFAuthenticationSession(url: url, callbackURLScheme: schemeURL) { [unowned self] url, _ in
+        #if canImport(AuthenticationServices)
```

Comment:
So turns out there was this handy pre-processor added into Swift 4.1 that let's us check for a Framework.
As this Framework does not exist in Xcode 9, it can not compile without this.

Reply:
These are two different kinds of checks, the `#if` is a pre-processor so if it's on Xcode 10+ include this code. If it was Xcode 9 only the code in the #else would actually be visible to the compiler. The `if  #available(iOS12)` is a runtime check, no escaping it.